src/api.rs

```diff
@@ -35,23 +46,109 @@ pub struct CronCat<'a, T: CronCatInterface> {
 }
 
 impl<'a, T: CronCatInterface> CronCat<'a, T> {
-    /// Swap assets in the cron_cat
+    /// Create task message
+    /// On success it will return [`croncat_integration_utils::CronCatTaskExecutionInfo`] in reply data
+    /// so you can save `task_hash` and `version` or any other useful information
     pub fn create_task(
         &self,
         task: CronCatTaskRequest,
+        funds: Vec<Coin>,
     ) -> AbstractSdkResult<CosmosMsg> {
-        todo!("logic that creates a task")
+        let modules = self.base.modules(self.deps);
+
+        let croncat_factory_address = modules.module_address(CRON_CAT_FACTORY)?;
```

Comment:
The module address function will get the address of a module that's installed on an account. In this case we want to expose interactions with the cron-cat module, so we should query the module with `CRONCAT_ID`. 

If we want the address of the cron_cat factory we can use the abstract name service contract query. You can do this by constructing an [`UncheckedContractEntry`](https://docs.rs/abstract-core/0.16.1/abstract_core/objects/struct.UncheckedContractEntry.html#method.check) and then checking it, which formats it, and then resolving it with the [`Resolve`](https://docs.rs/abstract-sdk/0.16.1/abstract_sdk/trait.Resolve.html) trait.